File: frontend/src/types/prediction.ts
export interface PredictionRequest {
location: string;
carpet_area_sqft: number;
floor_number: number; // was floor_num
bathroom: number;
balcony: number;
car_parking: number; // NEW
total_floors: number; // NEW
furnishing: string;
transaction: string;
ownership: string;
facing: string;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class PredictionRequest(BaseModel):
    carpet_area_sqft: float = Field(..., gt=0)

    floor_number: int = Field(..., ge=0)

    bathroom: int = Field(..., ge=0)

    balcony: int = Field(..., ge=0)

    car_parking: int = Field(..., ge=0)

    total_floors: int = Field(..., ge=1)

    furnishing: str = Field(..., min_length=1)

    transaction: str = Field(..., min_length=1)

    ownership: str = Field(..., min_length=1)

    facing: str = Field(..., min_length=1)


class PredictionResponse(BaseModel):
    predicted_price: float